mainbot.py

```diff
@@ -1,33 +1,110 @@
+import slack
+import asyncio
+import logging
 from slacker import Slacker
 from typing import List
 from datetime import datetime, timedelta
 
 
-def sort_messages(messages: List[dict], topk: int = 20) -> List[dict]:
+slacker: Slacker
+CRAWL_INTERVAL: int = 60 * 15  # in seconds
+
+
+async def crawl_messages() -> None:
+    def write_to_db_mock(anything):
+        print(anything, end="\n\n")
+
+    while True:
+        # get data
+        ch_info = await slacker.get_channels_list()
+        for ch_id, ch_name in ch_info:
+            print(f"Channel: {ch_name}")
+
+            day_ago = datetime.now() - timedelta(days=1)
+            messages = await slacker.get_channel_messages(ch_id, day_ago)
+            write_to_db_mock(messages)
+
+        # wait for next time
+        await asyncio.sleep(CRAWL_INTERVAL)
+
+
+def sort_messages(
+    messages: List[dict], topk: int = 20, criterion: str = "replies"
```

Comment:
I think It will be more convenient to use Enum for a criterion.

Reply:
will be on DB side